### DISCUSSION TITLE
Symfony 5.4 \Symfony\Component\Cache\LockRegistry bug?

### DISCUSSION THREAD

**Discussion by jan-egert** [symfony/symfony#44747] [Q&A] [answered]
Hello, I've migrated Symfony 5.3 app (working without any problem) to 5.4 and it seems there's some bug in \Symfony\Component\Cache\LockRegistry in combination with PHP-FPM. Looks like some kind of deadlock issue. FPM processes get into infinite loop state consuming 100% CPU (together) without any error/warning when app cache is used (tried redis and file adapter - both worked well in 5.3 with the same configuration) when under load (just 10 concurrent request using apache benchmark). During debugging I found FPM slow log mentioning repeatedly sem_acquire() (vendor/symfony/cache/LockRegistry.php:134) . Turning of sem_get function hasn't helped. Overwriting LockRegistry.php by the one from 5.3 helped. Has anyone found the same issue or is there some trick to avoid it in symfony and/or server configuration? If not, be careful - I spend a lot of time finding the problem because it works well in symfony server (using same php.ini) and pages not using app cache and not showing any errors.

**Comment by stof** [comment, marked as answer]
This will be fixed in the upcoming 5.4.2 and 6.0.2 releases thanks to https://github.com/symfony/symfony/pull/44667

**Reply by jan-egert** [reply]
Thanks. Do you know when the next release will be able to decide if to go back to 5.3 or wait?
